domain/src/DomainValidator.php

```diff
@@ -104,7 +105,7 @@ public function validate(DomainInterface $domain) {
       }
     }
     // Allow modules to alter this behavior.
-    \Drupal::moduleHandler()->invokeAll('domain_validate', $error_list, $hostname);
+    \Drupal::moduleHandler()->invokeAll('domain_validate', array($error_list, $hostname));
```

Comment:
On further thought, I believe we should do incremental PRs, rather than one big, does it all PR.
Thus fixing this hook call (depending on your preference on direct hook or alter hook) and proper tests should happen in its own little PR.